expyfun/stimuli/tests/test_tracker.py

```diff
@@ -72,11 +74,26 @@ def test_tracker_ud():
     tr = TrackerUD(None, 3, 1, [1, 0.5], [1, 0.5], 10, np.inf, 1,
                    change_indices=[2])
     tr.respond(True)
-    tr = TrackerUD(None, 3, 1, [1, 0.5], [1, 0.5], np.inf, 10, 1,
-                   change_indices=[2], change_rule='trials')
-    for t in np.arange(2):  # run long enough to encounter change_indices
-        tr.respond(True)
-        tr.respond(False)
+
+    with warnings.catch_warnings(record=True) as w:
+        tr = TrackerUD(None, 1, 1, 0.75, 0.25, np.inf, 7, 1,
+                       x_min=0, x_max=2)
+        responses = [True, False, False, False, True, True, True, False]
+        for r in responses:  # run long enough to encounter change_indices
+            tr.respond(r)
+        assert(tr.check_valid(1))  # make sure checking validity is good
+        assert(not tr.check_valid(3))
+        assert_equal(len(w), 1)
+        assert_raises(ValueError, tr.threshold)
```

Comment:
It's good to have this check, but more complete to have something a check that `tr.threshold(-2)` works (or whatever you need to do to use the last `N` reversals that are valid) and that the value is correct. (If there are no valid reversals at the end, modify the test so that there are a few valid ones at the end that you can use.)